static/templates/tab_bar.hbs

```diff
@@ -1,19 +1,24 @@
 <span id="tab_list">
     {{#if stream_narrow}}
-    <span {{#if stream_settings_link}}class="stream"{{/if}}>
-        {{#if icon}}
-        <i class="fa fa-{{icon}}" aria-hidden="true"></i>
+        {{#if stream_is_known}}
+            <span {{#if stream_settings_link}}class="stream"{{/if}}>
+                {{#if icon}}
+                <i class="fa fa-{{icon}}" aria-hidden="true"></i>
+                {{/if}}
+                <a href="{{stream_settings_link}}">{{title}}</a>
+            </span>
+            <span class="sub_count" data-toggle="tooltip" title="{{#tr this}}{{sub_count}} users are subscribed to #{{title}}{{/tr}}"><i class="fa fa-user-o"></i>{{formatted_sub_count}}</span>
+            {{#if rendered_narrow_description}}
+            <span class="narrow_description rendered_markdown" data-toggle="tooltip">{{rendered_markdown rendered_narrow_description}}</span>
+            {{else}}
+            <span class="narrow_description rendered_markdown" data-toggle="tooltip">{{#tr this}}(no description){{/tr}}</span>
+            {{/if}}
+        {{else}}
+        <i class="fa fa-question" aria-hidden="true"></i>
+        {{#tr this}}Unknown Stream{{/tr}}
+        <span class="sub_count" data-toggle="tooltip" title={{#tr this}}"0 subscribers to unknown stream{{/tr}}"><i class="fa fa-user-o"></i>{{#tr this}}0{{/tr}}</span>
+        <span class="narrow_description rendered_markdown" data-toggle="tooltip">{{#tr this}}This stream does not exist or is private.{{/tr}}</span>
```

Comment:
This refactor makes duplicate stream for translators and also duplicates various rendering details like the `sub_count` class; my guess is it's a regression because it'll mean folks can now accidentally make changes that only impact one of the two code paths.